gwpy/signal/spectral/_scipy.py

```diff
@@ -200,6 +206,35 @@ def coherence(timeseries, other, segmentlength, noverlap=None, **kwargs):
     --------
     scipy.signal.coherence
     """
+
+    # Should we warn about unequal sampling frequencies?
+    warn_fs = False
+
+    if downsample is None:
+        warn_fs = True
+        downsample = True
+
+    if timeseries.sample_rate != other.sample_rate:
+        # scipy assumes a single sampling frequency
+        if not downsample:
+            raise ValueError("Cannot calculate coherence when sampling "
+                             "frequencies are unequal")
+        else:
+            if warn_fs:
+                warnings.warn("Sampling frequencies are unequal. Higher "
+                              "frequency series will be downsampled before "
+                              "coherence is calculated",
+                              category=UserWarning)
+                # downsample the one with the higher rate
+                if timeseries.sample_rate > other.sample_rate:
+                    timeseries = timeseries.resample(other.sample_rate)
+                else:
+                    other = other.resample(timeseries.sample_rate)
+
+    # should never be equal from here on
+    if downsample is False:
+        assert timeseries.sample_rate == other.sample_rate
```

Comment:
Is this test necessary? If it should be kept, please add an explanatory comment (something along the lines of 'how did we get here?'

Also should the comment read 'should be equal from here on' (i.e. without the word 'never')?

Reply:
Yes, should have been "never" or "not". It's now not conditional on downsample. Put assertion in there for "defensive programming". Can remove if desired.